UnityProject/Assets/Scripts/NPC/MobSprite.cs

```diff
@@ -0,0 +1,177 @@
+
+		#region Lifecycle
+
+		private void OnEnable()
+		{
+			EnsureInit();
+		}
+
+		private void EnsureInit()
+		{
+			if (health != null) return;
+			health = GetComponent<LivingHealthBehaviour>();
+			directional = GetComponent<Directional>();
+			directionalSprite = GetComponent<DirectionalSpriteV2>();
+
+			directional.OnDirectionChange.AddListener(OnDirectionChange);
+		}
+
+		public override void OnStartServer()
+		{
+			EnsureInit();
+			spriteRot = 0;
+		}
+
+		public override void OnStartClient()
+		{
+			AddOverlayGameObjects();
+			EnsureInit();
+			SyncRotChange(spriteRot, spriteRot);
+		}
+
+		/// <summary>
+		/// Instantiate and attach the sprite overlays if they don't exist
+		/// </summary>
+		private void AddOverlayGameObjects()
+		{
+			if (burningOverlay == null)
+			{
+				burningOverlay = Instantiate(burningPrefab, transform).GetComponent<PlayerDirectionalOverlay>();
+				burningOverlay.enabled = true;
+				burningOverlay.StopOverlay();
+			}
+			if (electrocutedOverlay == null)
+			{
+				electrocutedOverlay = Instantiate(electrocutedPrefab, transform).GetComponent<PlayerDirectionalOverlay>();
+				electrocutedOverlay.enabled = true;
+				electrocutedOverlay.StopOverlay();
+			}
+		}
+
+		#endregion Lifecycle
+
+		//The local rotation of the sprite obj
+		private void SyncRotChange(float oldRot, float newRot)
+		{
+			EnsureInit();
+			spriteRot = newRot;
+			spriteRend.transform.localEulerAngles = new Vector3(0f, 0f, spriteRot);
+		}
+
+		/// <summary>
+		/// Sets the local rotation of the sprite obj
+		/// </summary>
+		/// <param name="newRot"></param>
+		public void SetRotationServer(float newRot)
+		{
+			spriteRot = newRot;
+		}
+
+		/// <summary>
+		/// Set the mob sprite to the first alive sprite.
+		/// </summary>
+		public void SetToAlive(bool network = false)
+		{
+			spriteHandler.ChangeSprite(aliveSpriteIndex, network);
+		}
+
+		/// <summary>
+		/// Set the mob sprite to the dead sprite.
+		/// </summary>
+		public void SetToDead(bool network = false)
+		{
+			if (spriteHandler == null)
+			{
+				Logger.LogError($"SpriteHandler missing on {gameObject}!");
+				return;
+			}
+
+			spriteHandler.ChangeSprite(deadSpriteIndex, network);
+		}
+
+		/// <summary>
+		/// Set the sprite renderer to bodies when the mob has died
+		/// </summary>
+		public void SetToBodyLayer()
+		{
+			spriteRend.sortingLayerName = "Bodies";
+		}
+
+		/// <summary>
+		/// Set the mobs sprite renderer to NPC layer
+		/// </summary>
+		public void SetToNPCLayer()
+		{
+			spriteRend.sortingLayerName = "NPCs";
+		}
+
+		public void ClearBurningOverlay()
+		{
+			if (burningOverlay == null)
+			{
+				return;
+			}
+
+			burningOverlay.StopOverlay();
+		}
+
+		public void SetBurningOverlay()
+		{
+			if (burningOverlay == null)
+			{
+				AddOverlayGameObjects();
+			}
+
+			burningOverlay.StartOverlay(directional.CurrentDirection);
+		}
+
+		private void OnDirectionChange(Orientation newDirection)
```

Comment:
Codacy found an issue: [Remove this unused method parameter 'newDirection'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6356172)

Reply:
`SyncVar`.

Reply:
This is an event. Cannot remove the parameter.